downloader_test.go

```diff
@@ -23,144 +23,156 @@ func TestDownloader(t *testing.T) {
 			})
 
-			it.After(func() {
-				err := os.Chdir(ogWd)
+			it("downloads from a 'file://' URI tgz", func() {
+				absPath, err := filepath.Abs(tgz)
+				h.AssertNil(t, err)
+
+				uri, err := paths.FilePathToUri(absPath)
 				h.AssertNil(t, err)
-			})
 
-			it("download from tgz", func() {
-				out, err := subject.Download(filepath.Base(tgz))
+				out, err := subject.Download(uri)
 				h.AssertNil(t, err)
 				h.AssertMatch(t, out, `\.tgz$`)
 				h.AssertOnTarEntry(t, out, "file.txt", h.ContentEquals("some file contents"))
 			})
-		})
-
-		it("download from an absolute directory", func() {
-			absPath, err := filepath.Abs(filepath.Join("testdata", "downloader", "dirA"))
-			h.AssertNil(t, err)
-
-			out, err := subject.Download(absPath)
-			h.AssertNil(t, err)
-			h.AssertNotEq(t, out, "")
-			h.AssertDirContainsFileWithContents(t, out, "file.txt", "some file contents")
-		})
-
-		it("download from an absolute tgz", func() {
-			absPath, err := filepath.Abs(tgz)
-			h.AssertNil(t, err)
-
-			out, err := subject.Download(absPath)
-			h.AssertNil(t, err)
-			h.AssertMatch(t, out, `\.tgz$`)
-			h.AssertOnTarEntry(t, out, "file.txt", h.ContentEquals("some file contents"))
-		})
-
-		it("download from a 'file://' URI directory", func() {
-			absPath, err := filepath.Abs(filepath.Join("testdata", "downloader", "dirA"))
-			h.AssertNil(t, err)
 
-			uri, err := paths.FilePathToUri(absPath)
-			h.AssertNil(t, err)
-
-			out, err := subject.Download(uri)
-			h.AssertNil(t, err)
-			h.AssertNotEq(t, out, "")
-			h.AssertDirContainsFileWithContents(t, out, "file.txt", "some file contents")
-		})
-
-		it("download from a 'file://' URI tgz", func() {
-			absPath, err := filepath.Abs(tgz)
-			h.AssertNil(t, err)
-
-			uri, err := paths.FilePathToUri(absPath)
-			h.AssertNil(t, err)
-
-			out, err := subject.Download(uri)
-			h.AssertNil(t, err)
-			h.AssertMatch(t, out, `\.tgz$`)
-			h.AssertOnTarEntry(t, out, "file.txt", h.ContentEquals("some file contents"))
-		})
+			it("downloads from a 'http(s)://' URI tgz", func() {
+				server := ghttp.NewServer()
+				server.AppendHandlers(func(w http.ResponseWriter, r *http.Request) {
+					http.ServeFile(w, r, tgz)
+				})
+				defer server.Close()
 
-		it("download from a 'http(s)://' URI tgz", func() {
-			server := ghttp.NewServer()
-			server.AppendHandlers(func(w http.ResponseWriter, r *http.Request) {
-				http.ServeFile(w, r, tgz)
+				out, err := subject.Download(server.URL() + "/downloader/somefile.tgz")
+				h.AssertNil(t, err)
+				h.AssertMatch(t, out, `\.tgz$`)
+				h.AssertOnTarEntry(t, out, "file.txt", h.ContentEquals("some file contents"))
 			})
-			defer server.Close()
 
-			out, err := subject.Download(server.URL() + "/downloader/somefile.tgz")
-			h.AssertNil(t, err)
-			h.AssertMatch(t, out, `\.tgz$`)
-			h.AssertOnTarEntry(t, out, "file.txt", h.ContentEquals("some file contents"))
-		})
+			it("caches to versioned directory", func() {
```

Comment:
This is the only added test. All other changes here are just clean up/organizational.